apps/omg_api/lib/state.ex

```diff
@@ -176,30 +167,37 @@ defmodule OMG.API.State do
   Also, eth_height given is the Ethereum chain height where the block being closed got submitted, to be used with events.
 
   Someday, one might want to skip some of computations done (like calculating the root hash, which is scrapped)
+
+  Returns `db_updates` due and relies on the caller to do persistence
   """
   def handle_call({:close_block, eth_height}, _from, state) do
-    {duration, {result, new_state}} = :timer.tc(fn -> do_form_block(state, eth_height) end)
-    _ = Logger.debug(fn -> "Closing block done in #{inspect(round(duration / 1000))} ms" end)
-    {:reply, result, new_state}
+    {:ok, child_block_interval} = Eth.RootChain.get_child_block_interval()
+
+    {:ok, {_block, event_triggers, db_updates}, new_state} = Core.form_block(child_block_interval, state)
+
+    # enrich the event triggers with the ethereum height supplied
+    event_triggers
+    |> Enum.map(&Map.put(&1, :submited_at_ethheight, eth_height))
+    |> EventerAPI.emit_events()
+
+    {:reply, {:ok, db_updates}, new_state}
   end
 
   @doc """
   Wraps up accumulated transactions submissions into a block, triggers db update and:
    - emits events to Eventer (if it is running, i.e. in Watcher).
    - pushes the new block into the respective service (if it is running, i.e. in Child Chain server)
    - enqueues the new block for submission to BlockQueue (if it is running, i.e. in Child Chain server)
+
+  Does its on persistence!
   """
   def handle_cast(:form_block, state) do
-    _ = Logger.debug(fn -> "Forming new block..." end)
-    {duration, {_result, new_state}} = :timer.tc(fn -> do_form_block(state) end)
-    _ = Logger.info(fn -> "Forming block done in #{inspect(round(duration / 1000))} ms" end)
-    {:noreply, new_state}
-  end
-
-  defp do_form_block(state, eth_height \\ nil) do
     {:ok, child_block_interval} = Eth.RootChain.get_child_block_interval()
 
-    {core_form_block_duration, {:ok, {%Block{number: blknum} = block, event_triggers, db_updates}, new_state}} =
+    _ = Logger.debug(fn -> "Forming new block..." end)
+
+    {core_form_block_duration,
+     {:ok, {%Block{number: blknum, hash: blkhash} = block, _event_triggers, db_updates}, new_state}} =
       :timer.tc(fn -> Core.form_block(child_block_interval, state) end)
```

Comment:
why :close_block don't print duration but :form_block print?

Reply:
for `form_block` (child chain) it is the only logs that would print on doing a new block. Also there are calculations that actually matter for the child chain (hashing).

In `close_block` (watcher) case these logs and time isn't relevant in my opinion

Reply:
:close_block and :form_block they are very similar, I thought about separating the common part.